opensquirrel/circuit_builder.py

```diff
@@ -41,40 +46,43 @@ class CircuitBuilder(GateLibrary, MeasurementLibrary):
     def __init__(
         self,
         qubit_register_size: int,
-        gate_set: [Callable[..., Gate]] = default_gate_set,
-        gate_aliases: Dict[str, Callable[..., Gate]] = default_gate_aliases,
-        measurement_set: List[Callable[..., Measure]] = default_measurement_set,
+        gate_set: list[Callable[..., Gate]] = default_gate_set,
+        gate_aliases: Mapping[str, Callable[..., Gate]] = default_gate_aliases,
+        measurement_set: list[Callable[..., Measure]] = default_measurement_set,
     ):
         GateLibrary.__init__(self, gate_set, gate_aliases)
         MeasurementLibrary.__init__(self, measurement_set)
         self.register_manager = RegisterManager(qubit_register_size)
         self.ir = IR()
 
-    def __getattr__(self, attr):
-        def add_comment(comment_string: str) -> CircuitBuilder:
-            self.ir.add_comment(Comment(comment_string))
-            return self
+    def __getattr__(self, attr: Any) -> Callable[..., Self]:
+        if attr == "comment":
+            return self._add_comment
+
+        return partial(self._add_instruction, attr)
 
-        def add_instruction(*args: tuple) -> CircuitBuilder:
-            if any(attr == measure.__name__ for measure in self.measurement_set):
-                generator_f = MeasurementLibrary.get_measurement_f(self, attr)
-                _check_generator_f_args(generator_f, args)
-                self.ir.add_measurement(generator_f(*args))
-            else:
-                generator_f = GateLibrary.get_gate_f(self, attr)
-                _check_generator_f_args(generator_f, args)
-                self.ir.add_gate(generator_f(*args))
-            return self
+    def _add_comment(self, comment_string: str) -> Self:
+        self.squirrel_ir.add_comment(Comment(comment_string))
```

Comment:
I thought this was fixed, wasn't it?
```suggestion
        self.ir.add_comment(Comment(comment_string))
```

Reply:
OK, I see they're gone now. Sorry about that!